app/src/main/kotlin/com/google/samples/apps/nowinandroid/navigation/NiaNavHost.kt

```diff
@@ -49,18 +49,29 @@ fun NiaNavHost(
         startDestination = ForYouBaseRoute,
         modifier = modifier,
     ) {
-        forYouSection(
-            onTopicClick = navController::navigateToTopic,
-        ) {
-            topicScreen(
-                showBackButton = true,
-                onBackClick = navController::popBackStack,
+        appState.niaNavigator.getNavigator(ForYouBaseNavigator::class.java).screen(
```

Comment:
Replace directly invoking navigation element with getting the requisite navigator instance, and invoking its `screen` method.